### ISSUE TITLE
Possible infinite loop

### ISSUE THREAD

**Issue by kcrisman** [openwebwork/webwork-open-problem-library#640]
In https://github.com/openwebwork/webwork-open-problem-library/blob/master/OpenProblemLibrary/ASU-topics/setStat/gust34.pg I have now had two reports of infinite loops.  It is possible this is in a previous version of the problem, but I don't have a way to check whether https://github.com/openwebwork/webwork-open-problem-library/commit/0914437ac334603bf1f15257757073ac63121956#diff-6869a7a2f81f6677f7b7e7123a38069f fixed it or not, though that seems possible - please close this if that is what that was intended to fix.

For reference, a seed number that caused problems is 2070; replacing it with 36 seems to have fixed the problem.

**Comment by jwj61** [comment]
Works for me with seed 2070, and do don't see how the current code would generate an infinite loop.  The randomization code in the problem has not changed for several years.  Has it been that long since you updated?

**Comment by kcrisman** [comment]
It's possible that it has been that long, yes.  We are running 2.12 - don't ask why we haven't upgraded, let's just say unfortunately WW upgrades have to be at a lower priority than a lot of other stuff the past same few years.

Did the "big commit" which is the last one for this file have something to do with preventing such loops?

**Comment by jwj61** [comment]
No, the one before that.

You can update the OPL without updating webwork.  It is pretty rare for a problem to stop working as a result.

Since this is fixed (as far as I can tell), I am closing the issue.

**Comment by maharper** [comment]
It looks like there was the possibility of an infinite loop, reported in bug #3819 in 2016.  http://bugs.webwork.maa.org/show_bug.cgi?id=3819
This was fixed on 2016-11-14 with commit 9a06f9d

**Comment by kcrisman** [comment]
I see, that was what that commit fixed!  I was wondering.  Thank you very much - I'll see if I can convince the sysadmin to update the OPL at least once the semester is done, though he has many other duties.  I'm sorry to keep posting these.

**Comment by jwj61** [comment]
No problem.  (When looking at a file on github, click on blame.  It will show what commit most recently changed each line.)
